### ISSUE TITLE
WPPF -- Rietveld refinement error

### ISSUE THREAD

**Issue by saransh13** [HEXRD/hexrdgui#1296]
Reproducibly getting the following error in the GUI when `Rietveld.Refine()` is run 3 times. The first 2 iterations run successfully and fits fine. No such error is thrown up if I run the program as a script in the terminal. Screenshots attached below.

```Traceback (most recent call last):
  File "/Users/saransh1/hexrdgui/hexrd/ui/calibration/wppf_runner.py", line 44, in run_wppf
    self.wppf_object = dialog.wppf_object
  File "/Users/saransh1/hexrdgui/hexrd/ui/calibration/wppf_options_dialog.py", line 647, in wppf_object
    self.update_wppf_object()
  File "/Users/saransh1/hexrdgui/hexrd/ui/calibration/wppf_options_dialog.py", line 724, in update_wppf_object
    setattr(obj, key, val)
  File "/Users/saransh1/hexrd/hexrd/wppf/WPPF.py", line 2582, in phases
    self.calctth()
  File "/Users/saransh1/hexrd/hexrd/wppf/WPPF.py", line 1771, in calctth
    t = self.phases[p][k].getTTh(l[0].value)
  File "/Users/saransh1/hexrd/hexrd/wppf/phase.py", line 901, in getTTh
    self.dsp = _calc_dspacing(self.rmt.astype(np.float64),
  File "/Users/saransh1/opt/miniconda3/envs/hexrd/lib/python3.9/site-packages/numba/core/dispatcher.py", line 468, in _compile_for_args
    error_rewrite(e, 'typing')
  File "/Users/saransh1/opt/miniconda3/envs/hexrd/lib/python3.9/site-packages/numba/core/dispatcher.py", line 409, in error_rewrite
    raise e.with_traceback(None)
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function getitem>) found for signature:
 
 >>> getitem(array(float64, 1d, C), Tuple(int64, slice<a:b>))
 
There are 22 candidate implementations:
  - Of which 20 did not match due to:
  Overload of function 'getitem': File: <numerous>: Line N/A.
    With argument(s): '(array(float64, 1d, C), Tuple(int64, slice<a:b>))':
   No match.
  - Of which 2 did not match due to:
  Overload in function 'GetItemBuffer.generic': File: numba/core/typing/arraydecl.py: Line 166.
    With argument(s): '(array(float64, 1d, C), Tuple(int64, slice<a:b>))':
   Rejected as the implementation raised a specific error:
     NumbaTypeError: cannot index array(float64, 1d, C) with 2 indices: Tuple(int64, slice<a:b>)
  raised from /Users/saransh1/opt/miniconda3/envs/hexrd/lib/python3.9/site-packages/numba/core/typing/arraydecl.py:88

During: typing of intrinsic-call at /Users/saransh1/hexrd/hexrd/wppf/xtal.py (17)

File "hexrd/hexrd/wppf/xtal.py", line 17:
def _calc_dspacing(rmt, hkls):
    <source elided>
    for ii in np.arange(nhkls):
        g = hkls[ii,:]
        ^
```

![Screen Shot 2022-09-29 at 1 08 52 PM](https://user-images.githubusercontent.com/15834451/193132554-c16030e1-a8d7-4ee7-a975-e2151dc7f05f.png)

![Screen Shot 2022-09-29 at 1 14 51 PM](https://user-images.githubusercontent.com/15834451/193132905-a7f18811-0590-4cc9-8de9-51ad27c11ba6.png)

**Comment by psavery** [comment]
Hi @saransh13. I was able to reproduce this, and I took a quick look at it. It looks like `self.hkls` [here](https://github.com/HEXRD/hexrd/blob/18cc1fb5e33a464c856b04c2e6f0d2b76d90a034/hexrd/wppf/phase.py#L223) is empty, and that's what is causing the error. Any idea why it is empty?

**Comment by psavery** [comment]
Actually, looking closer, it looks like the `hkls` is empty because all of the two theta values are `nan` for some reason. So I need to figure out why they are all `nan`.

**Comment by saransh13** [comment]
Aah yes, it should be in nm.

**Comment by psavery** [comment]
> Aah yes, it should be in nm. I'll make the change to wppf and create a PR.

Sounds good. I saw a couple of other places where it is just doing `.value` as well. Maybe we should make sure in all other places that the correct units are being converted to as well?

**Comment by saransh13** [comment]
Yes, will do a `grep` on all those cases.

**Comment by psavery** [comment]
Was this fixed by hexrd/hexrd#479?

**Comment by saransh13** [comment]
Yes, it should be (except the missing colon  on `line 92` in `RietveldHEDM.py` fixed in https://github.com/HEXRD/hexrd/pull/480

**Comment by psavery** [comment]
@joelvbernier We fixed it in hexrd/hexrd#480. It was a "wrong units" issue.
